### PR TITLE
feat: support /api/v3/secret endpoint in non-secure mode

### LINKED ISSUE #477: Implement `/api/v3/secrets` endpoint in non-secure mode
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
`device-sdk-go`

### Description [**REQUIRED**]
Currently, only secure mode supports the `/api/v2/secrets` endpoint. This means that both documentation and scripts need to supply the user with both options of storing secrets, one using the endpoint, and one using Consul InsecureSecrets directly.

It would be nice if that was abstracted away from the user such that the same endpoint can be called in secure and non-secure mode.

### Describe the solution you'd like
Calling `/api/v2/secrets` endpoint in non-secure mode will insert the provided secret into `InsecureSecrets` storage.

### Describe alternatives you've considered
- Manually documenting the 2 different ways of adding secrets
- Write scripts to abstract this functionality away from the user

### PR DESCRIPTION
**Pull request by ajcasagrande**
Implements the StoreSecret function for InsecureProvider. If the ConfigurationClient is in use, it will be used to store the secret, otherwise an error is returned.

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [X] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

**Run the new unit tests**
```
make test
```

**Try it out yourself**
1. clone latest edgex-compose
2. modify a device service to build using this code
go.mod:
```
replace github.com/edgexfoundry/go-mod-bootstrap/v3 => ../go-mod-bootstrap
```
3. build docker image
```
make docker
```
4. tag image so it is latest:
```
 docker tag edgexfoundry/device-usb-camera:0.0.0-dev nexus3.edgexfoundry.org:10004/device-usb-camera:latest 
```
5. pull latest edgex images
```
cd compose-builder
make pull
```
6. deploy in non-secure mode with `device-usb-camera`
```
make run no-secty ds-usb-camera
```
7. Wait for the service to start up
8. post a secret to to said device service:
```sh
curl --data '{
    "apiVersion":"v3",
    "secretName": "rtspauth",
    "secretData":[
        {
            "key":"username",
            "value":"<pick-a-username>"
        },
        {
            "key":"password",
            "value":"<pick-a-secure-password>"
        }
    ]
}' -X POST http://localhost:59983/api/v3/secret
```
9. expect a 201 Created
```
{"apiVersion":"v3","statusCode":201}
```
10. Verify the data is successfully in consul
http://localhost:8500/ui/dc1/kv/edgex/v3/device-usb-camera/Writable/InsecureSecrets

## New Dependency Instructions (If applicable)

Closes #477

### CODE REVIEW

**Comment by ajcasagrande** [comment]
recheck

File: bootstrap/secret/insecure.go [hunk 1]
```diff
@@ -26,6 +28,12 @@ import (
 	"github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger"
 )
 
+const (
+	// todo: should these be in `common` package?
+	writableKey        = "Writable"
```
Comment (lenny-intel) [inline bootstrap/secret/insecure.go:33]: Make the one here public and use it. 
https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/config/config.go#L52

File: bootstrap/secret/insecure.go [hunk 2]
```diff
@@ -102,8 +112,32 @@ func (p *InsecureProvider) GetSecret(secretName string, keys ...string) (map[str
...
+	// insert the top-level data about the secret name
+	basePath := fmt.Sprintf("%s/%s", insecureSecretsKey, secretName)
+	err := configClient.PutConfigurationValue(basePath+"/SecretName", []byte(secretName))
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindCommunicationError, "error setting secretName value in the config provider", err)
+	}
+
+	// insert each secret  key/value pair
+	for key, value := range secrets {
+		err = configClient.PutConfigurationValue(fmt.Sprintf("%s/SecretData/%s", basePath, key), []byte(value))
```
Comment (lenny-intel) [inline bootstrap/secret/insecure.go:132]: `SecretData` should be a constant also since it is also used in the unit tests.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: bootstrap/secret/insecure.go (hunk 1 above)
Comment (lenny-intel) [reply bootstrap/secret/insecure.go:33]: ok with the other InsecureSecrets specific constants being here.

**Review by lenny-intel — APPROVED** [review]
LGTM and TSC approved

**Pull request merged**